src/aideck/cache/cache_manager.py

```
@@ -0,0 +1,308 @@
+            session_id: Optional session identifier for cache isolation
+        """
+        self.session_id = session_id or f"session_{int(time.time())}"
+        self._cache_data: Dict[str, CacheEntry] = {}
+        self._cache_hits = 0
+        self._cache_misses = 0
+    
+    def _generate_cache_id(self, input_text: str) -> str:
+        """Generate a unique cache ID for the input text."""
+        if input_text is None:
+            input_text = ""
+        return hashlib.md5(input_text.encode()).hexdigest()
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)